src/components/modal/modal.tsx

```diff
@@ -0,0 +1,65 @@
+import { Component, ComponentInterface, h, Method, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'bds-modal',
+  styleUrl: 'modal.scss',
+  shadow: true,
+})
+export class BdsModal implements ComponentInterface {
+  /**
+   * Used to open/close the modal
+   */
+  @Prop({
+    mutable: true,
+    reflect: true,
+  })
+  public open?: boolean = false;
+
+    /**
+   * Used to open/close the modal
+   */
+  @Prop({
+    mutable: true,
+    reflect: true,
+  })
+  public closeButton?: boolean = true;
+
+  /**
```

Comment:
remove the code from line 27 to line 33, the @prop declaration and the comment should be above the property assignment.